### PR TITLE
upgrade geth to v1.10.15

### PR DESCRIPTION
**Pull request by crypto-facs**
Closes: #XXX

## Description

Upgrade Geth version to the latest stable release [v1.10.15](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.15). More detail explanation of changes can be found here https://github.com/ethereum/go-ethereum/milestone/127?closed=1

Major changes on the tracers were done on v.10.14 which required some changes on our side, specially around js tracer which requires a previous initiation of the tracer as a passive import on `app.go`. 

______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -93,7 +93,7 @@ endif
 ldflags += $(LDFLAGS)
 ldflags := $(strip $(ldflags))
 
-BUILD_FLAGS := -tags "$(build_tags)" -ldflags '$(ldflags)'
+-BUILD_FLAGS := -tags "$(build_tags)" -ldflags '$(ldflags)'
```
Comment (fedekunze) [inline Makefile:96]: why the change?

File: x/evm/keeper/grpc_query.go [hunk 2]
```diff
@@ -541,15 +543,15 @@ func (k *Keeper) traceTx(
 
 	// Depending on the tracer type, format and return the trace result data.
 	switch tracer := tracer.(type) {
```
Comment (fedekunze) [inline x/evm/keeper/grpc_query.go:545]: are we adding the native tracer (go)?

File: x/evm/keeper/state_transition.go [hunk 3]
```diff
@@ -107,11 +107,10 @@ func (k Keeper) VMConfig(ctx sdk.Context, msg core.Message, params types.Params,
 	}
 
 	return vm.Config{
-		Debug:       debug,
-		Tracer:      tracer,
-		NoRecursion: false, // TODO: consider disabling recursion though params
```
Comment (fedekunze) [inline x/evm/keeper/state_transition.go:112]: not required anymore?

**Review by fedekunze — APPROVED** [review]
ACK pending changelog

File: Makefile (hunk 1 above)
Comment (crypto-facs) [reply Makefile]: a mistake 🙏

File: x/evm/types/tracer.go [hunk 4]
```diff
@@ -21,22 +23,22 @@ const (
...
 // EVM transaction.
-func NewTracer(tracer string, msg core.Message, cfg *params.ChainConfig, height int64) vm.Tracer {
+func NewTracer(tracer string, msg core.Message, cfg *params.ChainConfig, height int64) vm.EVMLogger {
 	// TODO: enable additional log configuration
-	logCfg := &vm.LogConfig{
+	logCfg := &logger.Config{
 		Debug: true,
 	}
 
 	switch tracer {
```
[the remark this answers is not in the document]
Comment (crypto-facs) [reply x/evm/types/tracer.go:32]: native tracer falls under `tracers.Tracer` interface

**Review by loredanacirstea — APPROVED** [review]
LGTM - I did similar changes in my branch when trying to work with the latest development version of geth v.1.10.15
Caveat: I do not yet understand why some of the PR checks fail.

File: CHANGELOG.md [hunk 5]
```diff
@@ -55,6 +55,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (evm)  Fix `SelfDestruct` opcode by deleting account code and state.
 * (feemarket) [tharsis#855](https://github.com/tharsis/ethermint/pull/855) consistent `BaseFee` check logic.
 * (evm) [tharsis#729](https://github.com/tharsis/ethermint/pull/729) Refactor EVM StateDB implementation.
+* (evm) [tharsis#935](https://github.com/tharsis/ethermint/pull/935) Bumb Go-ethereum version to [`v1.0.15`](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.15)
```
Comment (hanchon) [inline CHANGELOG.md:58]:
```suggestion
* (evm) [tharsis#935](https://github.com/tharsis/ethermint/pull/935) Bumb Go-ethereum version to [`v1.10.15`](https://github.com/ethereum/go-ethereum/releases/tag/v1.10.15)
```
typo

Comment (crypto-facs) [reply CHANGELOG.md]: Great catch

File: x/evm/types/chain_config.go [hunk 6]
```diff
@@ -30,6 +30,8 @@ func (cc ChainConfig) EthereumConfig(chainID *big.Int) *params.ChainConfig {
 		MuirGlacierBlock:        getBlockValue(cc.MuirGlacierBlock),
 		BerlinBlock:             getBlockValue(cc.BerlinBlock),
 		LondonBlock:             getBlockValue(cc.LondonBlock),
+		ArrowGlacierBlock:       nil,
+		MergeForkBlock:          nil,
```
Comment (fedekunze) [inline x/evm/types/chain_config.go:34]: this should be `getBlockValue`

**Review by fedekunze — CHANGES_REQUESTED** [review]
Minor comment only

**Review by fedekunze — APPROVED** [review]

**Pull request merged**